File: tests/src/main/csharp/etch.tests/MainTypesListener.cs
// $Id$
// 
// Licensed to the Apache Software Foundation (ASF) under one
// or more contributor license agreements. See the NOTICE file
// distributed with this work for additional information
// regarding copyright ownership. The ASF licenses this file
// to you under the Apache License, Version 2.0 (the
// "License"); you may not use this file except in compliance
// with the License. You may obtain a copy of the License at
// 
//   http://www.apache.org/licenses/LICENSE-2.0
// 
// Unless required by applicable law or agreed to in writing,
// software distributed under the License is distributed on an
// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
// KIND, either express or implied. See the License for the
// specific language governing permissions and limitations
// under the License.
// 
using System;
using Org.Apache.Etch.Bindings.Csharp.Support;
using Org.Apache.Etch.Bindings.Csharp.Util;
using org.apache.etch.tests;

namespace etch.bindings.csharp.examples.types
{

///
///<summary> Main implementation for TypesListener. </summary>
///
public class MainTypesListener : TypesHelper.TypesServerFactory
{
	///
 	/// <summary>Main for TypesListener.</summary>
	/// <param name="args"></param>
	///
	public static void Main1(String[] args)
	{
		string uri = "tcp://localhost:4001";
		

		MainTypesListener implFactory = new MainTypesListener();
        ServerFactory listener = TypesHelper.NewListener(uri, null, implFactory);

		listener.TransportControl( TransportConsts.START_AND_WAIT_UP, 4000 );
	}

	///
	/// <summary>Constructs the MainTypesListener.</summary>
	///
	public MainTypesListener()
	{
		// Nothing to do.
	}

	///
	/// <summary>Return a new instance of TypesServer.</summary>
	/// @param client
	///
	public TypesServer NewTypesServer( RemoteTypesClient client )
	{
		return new ImplTypesServer( client );
	}
}

}

File: tests/src/main/java/org/apache/etch/tests/MainTypesListener.java
/* $Id$
 *
 * Copyright 2007-2008 Cisco Systems Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License"); you may not
 * use this file except in compliance with the License. You may obtain a copy
 * of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
 * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
 * License for the specific language governing permissions and limitations
 * under the License.
 */

package org.apache.etch.tests;

import org.apache.etch.bindings.java.support.ServerFactory;
import org.apache.etch.tests.RemoteTypesClient;
import org.apache.etch.tests.TypesHelper;
import org.apache.etch.tests.TypesServer;
import org.apache.etch.util.core.io.Transport;



/**
 * Main program for TypesServer. This program makes a listener to accept
 * connections from MainTypesClient.
 */
public class MainTypesListener implements TypesHelper.TypesServerFactory
{
	/**
	 * Main program for TypesServer.
	 * 
	 * @param args
	 * @throws Exception
	 */
	public static void main( String[] args ) throws Exception
	{
		String uri = "tcp://0.0.0.0:4001";
		
		Transport<ServerFactory> listener = TypesHelper.newListener( uri, null,
			new MainTypesListener() );

		listener.transportControl( Transport.START_AND_WAIT_UP, 4000 );
	}

	public TypesServer newTypesServer( RemoteTypesClient client )
	{
		return new ImplTypesServer( client );
	}
}
